src/azure/policyHelper.ts

```diff
@@ -139,6 +154,160 @@ function dividePolicyRequests(policyRequests: PolicyRequest[]) {
   return [definitionRequests, initiativeRequests, assignmentRequests];
 }
 
+async function assignRoles(assignmentRequests: PolicyRequest[], assignmentResponses: any[], roleAssignmentResults: PolicyResult[]) {
+  let roleRequests: RoleRequest[] = [];
+
+  // create map of definition id to definition for all definitions in code
+  let definitionsMap = new Map();
+  const definitions = getAllPolicyDefinitions();
+  definitions.forEach(definition => definitionsMap.set(definition.policyInCode.id, definition.policyInCode));
+
+  let pendingAssignments: any[] = [];
+
+  for (let index = 0; index < assignmentRequests.length; index++) {
+    const policyAssignment: any = assignmentResponses[index];
+
+    // We will assign roles only when assignmnet was created and has identity field has principalId in it.
+    if (isCreateOperation(assignmentRequests[index]) && policyAssignment.identity && policyAssignment.identity.principalId) {
+      // Now we need roleDefinitionIds. We will try to get it from repo else we will make azure api call later to get definition.
+      const definition = definitionsMap.get(policyAssignment.properties.policyDefinitionId);
+      if (definition) {
+        roleRequests.push(...getRoleRequest(definition, policyAssignment));
+      }
+      else {
+        pendingAssignments.push(policyAssignment);
```

Comment:
Can we refractor something on these lines?
```
assignRoles(... ){

	let createdAssignments = filterCreatedassignments(assignments);
	//parseRoleFromDef fetches ALL from Azure service, instead of merging REPO and Azure, lets fetch all from Azure
	//{
	//	"defId" : [
	//		   "roleId1","roleId2","roleId3"
	//		  ]
	// }
	let roleDefinitions = parseRoleFromDef(createdAssignments);
	let roleRequests = getRoleRequests(createdAssignments, roleDefinitions);
	return createRoles(roleRequests);
}
getRoleRequests(createdAssignments, roleDefinitions){
	roleRequests : any;
	for(assignment : createdAssignments){
		for(roleId : roleDefinitions.get( assignment.defId ) ){
			roleRequests.push(
			{
				scope: assignment.properties.scope,
       				roleAssignmentId: uuidv4(),
        			roleDefinitionId: roleId,
        			principalId: assignment.identity.principalId,
        			policyAssignmentId: assignment.id,
        			policyDefinitionId: assignment.defId
			}
		        );
		}	
	}
}